### DISCUSSION TITLE
Lesson 4: Import Error with solcx

### DISCUSSION THREAD

**Discussion by HoneyMustardRanchBBQ** [smartcontractkit/full-blockchain-solidity-course-py#156] [Q&A] [answered]
Hello! I'm having some issues when trying to run "python deploy.py" at exactly the 3:44:49 mark in the course! Here is a copy of my code in VScode:

`from solcx import compile_standard, install_solc

install_solc("0.6.0")

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)`

_______________________________________________________________________________

When I type "python deploy.py" into the terminal of VScode, I get this response in the terminal window:

"Traceback (most recent call last):
  File "deploy.py", line 1, in <module>
    import solcx
ImportError: No module named solcx"

_______________________________________________________________________________

I have installed solcx already, as well. When I try the pip install py-solc-x command in terminal I get this response in the terminal window:

"Requirement already satisfied: py-solc-x in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (1.1.0)
Requirement already satisfied: semantic-version<3,>=2.8.1 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from py-solc-x) (2.8.5)
Requirement already satisfied: requests<3,>=2.19.0 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from py-solc-x) (2.26.0)
Requirement already satisfied: charset-normalizer~=2.0.0 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from requests<3,>=2.19.0->py-solc-x) (2.0.6)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from requests<3,>=2.19.0->py-solc-x) (1.26.7)
Requirement already satisfied: certifi>=2017.4.17 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from requests<3,>=2.19.0->py-solc-x) (2021.5.30)
Requirement already satisfied: idna<4,>=2.5 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from requests<3,>=2.19.0->py-solc-x) (3.2)"

_______________________________________________________________________________

Just trying to figure out how to proceed from here! 
Thanks for taking the time to help!

**Comment by PatrickAlphaC** [comment, marked as answer]
Could you please check the discussion here? https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/5#discussion-3566046

Marking as answer for now, let me know if that didn't work.

**Reply by grizz38** [reply]
Hi Patrick, i even tried to instal env for the solcx and it still giving me the same import error .. Please Help
<img width="872" alt="Screen Shot 2021-12-08 at 12 38 39 AM" src="https://user-images.githubusercontent.com/86847067/145123686-94505917-c57c-4f2f-a8b3-7a5810783b9d.png">

**Reply by cromewar** [reply]
@grizz38 does this: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/510#discussioncomment-1762842 works for you?

**Reply by alexpost81** [reply]
there are no solutions in the suggested discussion

**Reply by alexpost81** [reply]
> Could you please check the discussion here? [#5 (comment)](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/5#discussion-3566046)
> 
> Marking as answer for now, let me know if that didn't work.

this has not been answered

**Reply by cromewar** [reply]
@alexpost81  this might be helpful for you:  https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4

**Reply by alexpost81** [reply]
> @alexpost81 this might be helpful for you: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4

already been through all of those, not a single workable solution.

**Reply by alexpost81** [reply]
that doesn't work, I have yet to find a single working solution

**Reply by grizz38** [reply]
@alexpost81 Same, vscode cant seem to recognize module solc-x I've basically tried everything

**Reply by alexpost81** [reply]
ok, my VS code just asked to restart and install an update and it works now. Sorry I didn't see what it exactly said, but check for updates, it may work

**Reply by francesco12801** [reply]
I've the same problem but it's still no working

**Reply by francesco12801** [reply]
can you help me?

**Reply by cromewar** [reply]
@francesco12801 could you pleas offer more info about the error and the thing you already tried?

**Reply by francesco12801** [reply]
thanks for the availability but I solved the problem :)

**Reply by alexpost81** [reply]
> thanks for the availability but I solved the problem :)

how did you solve it?

**Reply by jt-berg-whitt** [reply]
how did you solve? the other answers from people in the other threads arent working for me and many others in the past week

**Reply by selinimsibiri** [reply]
i tried to change my python select interpere i also uninstalled solc and installed again. Also tried installing 1.1.1 and 1.1.0 versions but still im taking this error:

Traceback (most recent call last):
  File "deploy.py", line 1, in <module>
   from solcx import compile_standard
ImportError: No module named solcx

can you please help me??
and here is my code:


  from solcx import compile_standard
from solcx.install import install_solc

# from solcx.install import SOLCX_BINARY_PATH_VARIABLE

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

    # Compile Our Solidity
    install_solc("0.6.0")

    compiled_sol = compile_standard(
        {
            "language": "Solidty",
            "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
            "settings": {
                "outputSelection": {
                    "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
                }
            },
        },
        solc_version="0.6.0",
    )

    import solcx

    solcx.install_solc("0.8.7")

    print(compiled_sol)

**Reply by iamade** [reply]
what solved the issue for me was using pip3 install py-solc-x instead of pip install py-solc-x

**Comment by semsion** [comment]
I'm running Ubuntu 20.04, and as Python 2 is installed, running 'python' in the terminal starts the Python 2 interpreter, which doesn't locate the installed solcx package. 

If you have a similar configuration you can resolve this by using 'python3' instead, or setting python 3 as default. 

Discussion here: https://askubuntu.com/questions/1232996/my-default-python-version-is-2-7-18rc1-after-upgrade-from-18-04-to-20-04

**Reply by cromewar** [reply]
I highly recommend to use python 3 instead.

**Reply by slime956** [reply]
pip install py-solc

**Reply by DarrenWayn** [reply]
https://stackoverflow.com/questions/69819421/importerror-no-module-named-solcx

python3 -m venv env 
source env/bin/activate
pip3 install py-solc-x 

this work for me

**Comment by abmi108** [comment]
I uninstalled and re-installed python. This did the trick for me.

**Reply by slime956** [reply]
did you uninstall all the packages as well, honestly i gave up around the 4 hour mark, but i remember having surpassed this part, i remember having solved the error message that "solcx could not be resolved/imported" , and i dont remember how i surpassed it. i uninstalled it and ima try from scratch, hopefully it works.
